server/lorax_server/models/custom_modeling/flash_qwen2_modeling.py

```diff
@@ -496,7 +495,64 @@ def forward(
             prefill_cache_indices,
             adapter_data,
         )
+
         if lm_head_indices is not None:
             hidden_states = hidden_states[lm_head_indices]
         logits, speculative_logits = self.lm_head(hidden_states, adapter_data)
         return logits, speculative_logits
+
+
+class FlashQwen2ForEmbeddings(torch.nn.Module):
+    def __init__(self, config, weights):
+        super().__init__()
+        self.config = config
+
+        self.model = FlashQwen2Model(config, weights)
+        self.max_past = config.sliding_window
+        self.output_weight = weights.get_tensor("linear.weight")
+        self.output_bias = weights.get_tensor("linear.bias")
+        # To satisfy the parent class interface
+        # TODO: fix
+        self.lm_head = None
```

Comment:
Shouldn't be needed at all no? What's the parent class here?

Reply:
The FlashQwen2 class' parent FlashCasualLm has a call to model.lm_head and so it breaks if this is not defined

Reply:
I see, that's interesting